platform/core/src/services/HangingProtocolService/HPMatcher.js

```diff
@@ -29,27 +29,31 @@ const match = (
   let requiredFailed = false;
   let score = 0;
 
+  // Allow for matching against current or prior specifically
+  const prior = options?.studies?.[1];
+  const current = options?.studies?.[0];
+  const instance = (metadataInstance.images || metadataInstance.others)?.[0];
+  const fromSrc = {
+    prior,
+    current,
+    instance,
+    ...options,
+    options,
+    metadataInstance,
+  };
+
   rules.forEach(rule => {
-    const { attribute } = rule;
+    const { attribute, from = 'metadataInstance' } = rule;
     // Do not use the custom attribute from the metadataInstance since it is subject to change
     if (customAttributeRetrievalCallbacks.hasOwnProperty(attribute)) {
       readValues[attribute] = customAttributeRetrievalCallbacks[
         attribute
-      ].callback(metadataInstance, options);
+      ].callback.call(rule, metadataInstance, options);
```

Comment:
this will break others `customAttributeRetrievalCallbacks`, can't you add it at the end?

Reply:
How?  This just means that IF the attribute is defined as a function, it will assign this as the rule that the callback comes from, which is the expected behaviour on calling functions.  It doesn't break anything.